packages/flame/lib/src/components/core/component.dart

```diff
@@ -543,17 +547,23 @@ class Component {
     return Future.wait(futures);
   }
 
-  /// Adds this component as a child of [parent] (see [add] for details).
-  FutureOr<void> addToParent(Component parent) {
+  FutureOr<void> _addChildImpl(Component child) {
     assert(
-      _parent == null,
-      '$this cannot be added to $parent because it already has a parent: '
-      '$_parent',
+      child._parent == null,
+      '$child cannot be added to $this because it already has a parent: '
+      '${child._parent}',
     );
-    _parent = parent;
-    parent.lifecycle._children.add(this);
-    if (!isLoaded && (parent.findGame()?.hasLayout ?? false)) {
-      return _startLoading();
+    child._parent = this;
+    final game = findGame();
+    if (isMounted) {
+      assert(game != null);
+      game!.fcsRoot!.enqueueAdd(child, this);
```

Comment:
do we need the assert if we are doing `!` anyway?
and if so do we need an assert on fcsRoot too?

Reply:
good point